### PR TITLE
Default sortMethod (plus: initialState from extra props)

### PR DESCRIPTION
**Pull request by dahlbyk**
## Griddle major version

1.x

## Changes proposed in this pull request

- Type cleanup for `RowRenderProperties` and `ColumnRenderProperties`, which currently the same as `RowDefinitionProps` and `ColumnDefinitionProps`.
- Rearranges how `initialState` is initialized to accept/override `initialState` from extra `Griddle` props. Everything still seems to work as expected, but please double-check this.
   * As a bonus, this allows setting `enableSettings` and `textProperties` without a plugin, as demonstrated with new stories.
- Default `sortMethod` from state is used, if it exists.
- Addressed TODO to get Next/Previous label from `textProperties`

## Why these changes are made

`<Griddle sortMethod={...}>` now works as a high-level hook to override all columns' sort behavior. It does not change default behavior (cc #466).

## Are there tests?

Stories!

### CODE REVIEW

**Comment by ryanlanciaux** [comment]
Ahh shoot -- didn't mean to make conflicts here.

**Comment by dahlbyk** [comment]
> Ahh shoot -- didn't mean to make conflicts here.

It's fine, I knew it was coming.

**Comment by dahlbyk** [comment]
(Conflicts have been resolved.)

File: stories/index.tsx [hunk 1]
```diff
@@ -178,6 +178,19 @@ storiesOf('Griddle main', module)
     )
   })
 
+  .add('with custom default sort', () => {
+    return (
```
Comment (ryanlanciaux) [inline stories/index.tsx]: Should this have the sortProperties set like the previous story?

```
const sortProperties = [
  { id: 'name', sortAscending: true }
];
...
<Griddle
  ...
  sortProperties={sortProperties}
  sortMethod={sortBySecondCharacter}>
```

**Comment by ryanlanciaux** [comment]
This looks great! I've run this locally on some of my projects also and everything seems to work well. I just have the one question on the story but it looks good to merge.

File: stories/index.tsx (hunk 1 above)
Comment (dahlbyk) [inline stories/index.tsx]: I don't think this grid needs to be pre-sorted, no. This test is derived from the [test that follows](https://github.com/NewBoCo/Griddle/blob/7ecddf1ec5ab8d933bb2fbc544788c251f691829/stories/index.tsx#L194-L206).

Comment (ryanlanciaux) [inline stories/index.tsx]: Got it - sorry about my confusion. 👍

**Pull request merged**